### PR TITLE
feat: switch to quad trees in stacked

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: storage-proofs/src/stacked/proof.rs [hunk 1]
```diff
@@ -388,12 +393,30 @@ impl<'a, H: 'static + Hasher, G: 'static + Hasher> StackedDrg<'a, H, G> {
...
-        let mut tree_r_last_config =
-            StoreConfig::from_config(&config, CacheKey::CommRLastTree.to_string(), None);
-        let mut tree_c_config =
-            StoreConfig::from_config(&config, CacheKey::CommCTree.to_string(), None);
+        let mut tree_d_config = StoreConfig::from_config(
+            &config,
+            CacheKey::CommDTree.to_string(),
+            Some(StoreConfig::default_cached_above_base_layer(
+                data.len(),
+                BINARY_ARITY,
```
Comment (cryptonemo) [inline storage-proofs/src/stacked/proof.rs]: Shouldn't all of the tree arities be matching?

File: storage-proofs/src/stacked/proof.rs [hunk 2]
```diff
@@ -454,7 +477,7 @@ impl<'a, H: 'static + Hasher, G: 'static + Hasher> StackedDrg<'a, H, G> {
                 trace!("building merkle tree for the original data");
                 data.ensure_data()?;
                 measure_op(CommD, || {
-                    Self::build_tree::<G>(data.as_ref(), tree_d_config.clone())
+                    Self::build_binary_tree::<G>(data.as_ref(), tree_d_config.clone())
```
Comment (cryptonemo) [inline storage-proofs/src/stacked/proof.rs]: I see that it's not here, just want to be sure it's intentional.

File: storage-proofs/src/stacked/proof.rs (hunk 1 above)
Comment (dignifiedquire) [reply storage-proofs/src/stacked/proof.rs]: no, comm_d will stay a binary tree

Comment (dignifiedquire) [reply storage-proofs/src/stacked/proof.rs]: maybe will change

File: storage-proofs/src/stacked/proof.rs (hunk 2 above)
Comment (dignifiedquire) [reply storage-proofs/src/stacked/proof.rs]: yes it is for now

Comment (cryptonemo) [reply storage-proofs/src/stacked/proof.rs]: Ok, thanks

**Pull request merged**